dl1_data_handler/image_mapper.py

```diff
@@ -648,6 +657,89 @@ def _get_grids(
         return input_grid, output_grid
 
 
+class HexagonalPatchMapper(ImageMapper):
+    """
+    HexagonalPatchMapper retrieves the necessary information to perform indexed 
+    convolutions, also allows croping the images following the "sipm_patches.h5"
+    patches geometry and reorders the pixels.
+
+    This class extends the functionality of ImageMapper by implementing
+    methods to look up at the configuration file and perform the image cropping.
+    It is particularly useful for applications where we are working with waveforms
+    with high time dimension.
+    """
+
+    def __init__(
+        self,
+        geometry,
+        config=None,
+        parent=None,
+        **kwargs,
+    ):
+        super().__init__(
+            geometry=geometry,
+            config=config,
+            parent=parent,
+            **kwargs,
+        )
+
+        if geometry.pix_type != PixelShape.HEXAGON:
+            raise ValueError(
+                f"HexagonalPatchMapper is only available for hexagonal pixel cameras. Pixel type of the selected camera is '{geometry.pix_type}'."
+             )
+
+        path = files("dl1_data_handler.ressources").joinpath("newcam_sipm_patches.h5")
```

Comment:
This file needs a versioning and a proper naming structure. I'm proposing 
triggerpatches_AdvCam_v1.h5

Reply:
If you rename "`UNKNOWN-7987PX`" earlier to AdvCam, you could retrieve this file via `f"{triggerpatches_{geometry.name}_v1}"`. Also the version should be a configuration. At this stage of the project we might want to experiment with different patches, so we need to be able to switch between different version without reinstalling the code.